packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_controller.dart

```diff
@@ -7,51 +7,145 @@ import 'package:flutter/foundation.dart';
 import 'deep_links_model.dart';
 import 'fake_data.dart';
 
-class DeepLinksController {
-  bool get showSpitScreen => showSpitScreenNotifier.value;
+enum SchemeFilterOption {
+  http,
+  custom,
+  showAll,
+}
 
-  List<LinkData> get getLinkDatasByPath {
-    final linkDatasByPath = <String, LinkData>{};
-    for (var linkData in linkDatasNotifier.value) {
-      linkDatasByPath[linkData.path] = linkData;
-    }
-    return linkDatasByPath.values.toList();
-  }
+enum OsFilterOption {
+  android,
+  ios,
+  showAll,
+}
 
-  List<LinkData> get getLinkDatasByDomain {
-    final linkDatasByDomain = <String, LinkData>{};
-    for (var linkData in linkDatasNotifier.value) {
-      linkDatasByDomain[linkData.domain] = linkData;
-    }
-    return linkDatasByDomain.values.toList();
-  }
+enum StatusFilterOption {
+  noIssue,
+  failedDomainCheck,
+  failedPathCheck,
+  failedAllCheck,
+  showAll,
+}
+
+class DeepLinksController {
+  bool get showSplitScreen => showSplitScreenNotifier.value;
+
+  List<LinkData> get getLinkDatasByPath =>
+      _getFilterredLinks(linkDatasByPath, _searchContentNotifier.value);
+  List<LinkData> get getLinkDatasByDomain =>
+      _getFilterredLinks(linkDatasByDomain, _searchContentNotifier.value);
 
   final selectedLink = ValueNotifier<LinkData?>(null);
   final linkDatasNotifier = ValueNotifier<List<LinkData>>(allLinkDatas);
-  final showSpitScreenNotifier = ValueNotifier<bool>(false);
-
+  final showSplitScreenNotifier = ValueNotifier<bool>(false);
+  final domainErrorCountNotifier = ValueNotifier<int>(0);
+  final pathErrorCountNotifier = ValueNotifier<int>(0);
   final _searchContentNotifier = ValueNotifier<String>('');
+  final schemeFilterOptionNotifier =
+      ValueNotifier<SchemeFilterOption>(SchemeFilterOption.showAll);
+  final osFilterOptionNotifier =
+      ValueNotifier<OsFilterOption>(OsFilterOption.showAll);
+  final statusFilterOptionNotifier =
+      ValueNotifier<StatusFilterOption>(StatusFilterOption.showAll);
 
-  void _updateLinks() {
-    final searchContent = _searchContentNotifier.value;
-    final List<LinkData> linkDatas = searchContent.isNotEmpty
-        ? allLinkDatas
-            .where(
-              (linkData) => linkData.matchesSearchToken(
-                RegExp(
-                  searchContent,
-                  caseSensitive: false,
-                ),
-              ),
-            )
-            .toList()
-        : allLinkDatas;
+  var linkDatasByDomain = <LinkData>[];
```

Comment:
Do we really want to cache the value here? it seems not worth the maintenance cost...